faiss/gpu/impl/IVFBase.cu

```diff
@@ -677,9 +680,209 @@ int IVFBase::addVectors(
             listOffsetDevice,
             stream);
 
+    // save index mapping
+    HostTensor<Index::idx_t, 1, true> indicesHost(indices, stream);
+    indexMapping(listIdsHost, listOffsetHost, indicesHost);
```

Comment:
same thing here, should be gated based on whether or not this feature is enabled upon index construction